src/plugins/integrations/HttpIntegration.ts

```diff
@@ -48,73 +51,77 @@ class HttpIntegration implements Integration {
   wrap(lib: any, config: any): void {
     function wrapper(request: any) {
       return function requestWrapper(options: any, callback: any) {
+        try {
+          const tracer = ThundraTracer.getInstance();
+          const method = (options.method || 'GET').toUpperCase();
+          options = typeof options === 'string' ? url.parse(options) : options;
+          const host = options.hostname || options.host || 'localhost';
+          const path = options.path || options.pathname || '/';
+          const queryParams = path.split('?').length > 1 ? path.split('?')[1] : '';
+
+          if (!HttpIntegration.isValidUrl(host)) {
+            return request.apply(this, [options, callback]);
+          }
 
-        const tracer = ThundraTracer.getInstance();
-        const method = (options.method || 'GET').toUpperCase();
-        options = typeof options === 'string' ? url.parse(options) : options;
-        const host = options.hostname || options.host || 'localhost';
-        const path = options.path || options.pathname || '/';
-        const queryParams = path.split('?').length > 1 ? path.split('?')[1] : '';
+          const parentSpan = tracer.getActiveSpan();
+          const span = tracer._startSpan(host + path, {
+            childOf: parentSpan,
+            domainName: DomainNames.API,
+            className: ClassNames.HTTP,
+            disableActiveStart: true,
+          });
+
+          if (!(Utils.getConfiguration(envVariableKeys.DISABLE_SPAN_CONTEXT_INJECTION) === 'true')) {
+            const headers = options.headers ? options.headers : {};
+            tracer.inject(span.spanContext, opentracing.FORMAT_TEXT_MAP, headers);
+            options.headers = headers;
+          }
 
-        if (!HttpIntegration.isValidUrl(host)) {
+          span.addTags({
+            [SpanTags.OPERATION_TYPE]: 'CALL',
```

Comment:
I think, anymore we can set operation type to the `method` here instead of `CALL` constant.

Reply:
Done.